### PR TITLE
Adds Torchscript-enabled BPE tokenizers

### PR DESCRIPTION
**Pull request by geoffreyangus**
This PR adds the final two tokenizers officially supported by torchtext, `CLIPTokenizer` and `GPT2BPETokenizer`.

The primary challenge in this work is that these two tokenizers behave slightly different than all other tokenizers supported by Ludwig. Out of the box, they return a `List[List[str]]` of _token ids_ directly (e.g. `[['23', '54', '76'], ['98', '1']]`). The rest of the Ludwig ecosystem expects tokenizers to return a `List[List[str]]` of tokens (e.g. `[['quick', 'brown', 'fox'], ['jump', 'ed']]`).

I work around this issue by making `vocab: List[str]` (which is passed to the tokenizer as a filepath) a tokenizer attribute, similar to [HFTokenizer](https://github.com/ludwig-ai/ludwig/blob/8309bc8f5ee8a92244ab25b2bf8ab70e9fdaa778/ludwig/utils/strings_utils.py#L241). During the forward pass, `vocab` is used to convert the token ids into the original string tokens. This vocab is also used as the vocab in `strings_utils.create_vocabulary`.

### CODE REVIEW

File: ludwig/utils/strings_utils.py [hunk 1]
```diff
@@ -258,7 +258,8 @@ def create_vocabulary(
             vocab = [padding_symbol] + vocab
         else:
             padding_symbol = pad_token
-
+    elif tokenizer_type in {"clip_tokenizer", "gpt2bpe_tokenizer"}:
```
Comment (tgaddair) [inline ludwig/utils/strings_utils.py:261]: Perhaps we can make this a little more generic:

```
elif hasattr(tokenizer, "get_vocab"):
    vocab = tokenizer.get_vocab()
```

File: tests/integration_tests/test_torchscript.py [hunk 2]
```diff
@@ -228,6 +226,7 @@ def test_torchscript_e2e(csv_filename, tmpdir):
         # set_feature(vocab_size=3),
         # bag_feature(vocab_size=3),
     ]
+    input_features += torchscript_enabled_text_features
```
Comment (tgaddair) [inline tests/integration_tests/test_torchscript.py:229]: Nit: more pythonic would be:

```
input_features = [
    ...,
    *torchscript_enabled_text_features
]
```

**Review by tgaddair — COMMENTED** [review]
Looks great! Just minor nits.

File: ludwig/utils/strings_utils.py (hunk 1 above)
Comment (geoffreyangus) [reply ludwig/utils/strings_utils.py:261]: Done.

File: tests/integration_tests/test_torchscript.py (hunk 2 above)
Comment (geoffreyangus) [reply tests/integration_tests/test_torchscript.py:229]: Done.

**Review by tgaddair — APPROVED** [review]

**Pull request merged**